### DISCUSSION TITLE
Plus Remote for Volume Reconstruction

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1172] [Q&A] [answered]
I'm trying to do a volume reconstruction of the spine phantom, I'm using Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB. I've tried with Plus Remote but when I do the Scout Scan the volume doesn't appear and a 'X' is marked in the section. What can I do to resolve it? Is there another way to the volume reconstruction?
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/8743394b-43ef-462e-a0a9-a5218a2b86c3)

**Comment by Sunderlandkyl** [comment]
Can you post your config file?

**Reply by LeidyMoraV** [reply]
https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/ERG61TyE5thFirSLZIaiHiMBOHQhZmOQ9clCpB_bgD_v1w?e=zD886C

**Reply by Sunderlandkyl** [reply]
The config file needs to have a "VirtualVolumeReconstructor" device.
See examples at the bottom of this page: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceVirtualVolumeReconstructor.html.

**Reply by LeidyMoraV** [reply]
Thanks! I added the Device configuration, but it does not work the 'X' shows:
`Timeout while waiting for volume reconstruction result`
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/2d506b9e-965f-4206-8780-ba25c843f3ac)

**Reply by Sunderlandkyl** [reply]
Is there an error in Plus?

**Reply by Sunderlandkyl** [reply]
Also can you upload the new config file as well?

**Reply by LeidyMoraV** [reply]
I don't see any error show in Plus. Here is the new config file: https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/ERW-yRIfoAlFo3LOezcy9JcBgu4SVOPX3kXPeAkmVM_y8Q?e=dgt1yt

I've tried using a different phantom (fcal 2.0) instead of the spine phantom and the result is the same. Do you think is maybe related to the fact that the US image contains more than the image in itself? I mean there is the User Interface too.
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/507fc118-772d-407f-8a18-f838efd38663)

**Reply by Sunderlandkyl** [reply]
Do you have a full PlusServer log?

I don't think the fact that the UI is shown will make a difference, but you can clip the image using "ClipRectangleOrigin " and "ClipRectangleSize" in the Epiphan device. See:

http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceEpiphan.html

**Reply by LeidyMoraV** [reply]
Should I use in the config file for the VolumeReconstructorDevice:
`ImageCoordinateFrame="Image" ReferenceCoordinateFrame="Reference"`
or
`ImageCoordinateFrame="Image" ReferenceCoordinateFrame="Image"` ?

Also here is the PlusServerLog, I think the error is related to admin access maybe?

![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/bfd1bc4e-8f39-40b4-aed1-b8984a7d6ae1)

**Reply by Sunderlandkyl** [reply]
ImageCoordinateFrame="Image" ReferenceCoordinateFrame="Reference" is correct.

Can you try specifying another folder in which you have write access?

By the way, if you just want to do the volume reconstruction fully within Slicer, you can have a look at the "Volume Reconstruction" module in SlicerIGT: https://www.youtube.com/watch?v=2vXeJxYFou4.

**Reply by LeidyMoraV** [reply]
I've tried following the tutorial but Slicer stops working and shows this error:
```
Slicer has caught an application error, please save your work and restart.

The application has run out of memory. Increasing virtual memory size in system settings or adding more RAM may fix this issue.
If you have a repeatable sequence of steps that causes this message, please report the issue following instructions available at https://slicer.org

The message detail is:
Exception thrown in event: bad allocation
```

![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/af62e069-6eb3-484e-8539-58b2db2eb02c)

**Reply by Sunderlandkyl** [reply]
I think the issue is that the ROI is too large, but it's hard to tell based on the image alone. How big did you make the ROI?

**Reply by LeidyMoraV** [reply]
I size down the ROI and the error resolved. But when I do the Volume Reconstruction no volumes appear, do you think its related to the calibration? Maybe the RAM memory? What else could affect the reconstruction?

**Reply by Sunderlandkyl** [reply]
It's probably either related to display visibility or issues with the the volume reconstruction itself.

If you share the mrb I can take a look at the scene.

**Reply by LeidyMoraV** [reply]
Thank you, here is the mrb file: 
https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/ESN_ky_eUyRKovb72i_OUg0Bqlf2EFt26mplHU9gh09lpA?e=vJMrdB

**Reply by Sunderlandkyl** [reply, marked as answer]
Looks like it's just a display issue. If you switch to the volume rendering module you can select the output volume, click on the eye to show it and then choose "MR-Default" in the "Preset" selector.

![image](https://github.com/PlusToolkit/PlusLib/assets/9222709/190e059d-b58f-4ab6-9bb9-32418a6d461e)

**Reply by LeidyMoraV** [reply]
Thank you so much for your help! I think in order to get the best volume reconstruction I need to modify the "ClipRectangleOrigin" so that's going to be my next step :)
